### ISSUE TITLE
DDLDatabase fails with order in Index statements

### ISSUE THREAD

**Issue by bernstein82** [jOOQ/jOOQ#7203]
Generating Records from sql file with DDLDatabase not working with ordered indexes:
### Actual behavior

```sql
CREATE UNIQUE INDEX A7IX019 ON A7IY019 (MI0149A ASC);
```

fails with :

```
Caused by: org.jooq.impl.ParserException: C42000_NO_SUBCLASS: Expected tokens: []:   ON A7IY019 (MI0149A [*]ASC);
``` 

### Expected behavior
just works

### Versions:
- jOOQ: 3.10.1
- Java: 1.8.144

**Comment by lukaseder** [comment]
Thanks for reporting this. Indeed, this is an oversight. The jOOQ API allows for `OrderField` to be used in `CREATE INDEX` statements, so the parser should allow it as well.

By the way, there have been quite a few parser improvements in jOOQ 3.10.5.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.11.0. Will merge to 3.10.6 (#7205)
